src/codegen/general.rs

```diff
@@ -16,32 +16,71 @@ use std::{
 
 pub fn start_comments(w: &mut dyn Write, conf: &Config) -> Result<()> {
     if conf.single_version_file.is_some() {
-        return start_comments_no_version(w);
+        return start_comments_no_version(w, conf);
     }
-    writeln!(
+    write!(
         w,
         "// This file was generated by gir (https://github.com/gtk-rs/gir @ {})
-// from gir-files (https://github.com/gtk-rs/gir-files @ {})
-// DO NOT EDIT",
-        VERSION, conf.girs_version
+{}// DO NOT EDIT",
+        VERSION,
+        conf.girs_version
+            .iter()
+            .map(|info| {
+                format!(
+                    "// from {} ({}{})\n",
+                    info.gir_dir.display(),
+                    match info.get_repository_url() {
+                        Some(url) => format!("{} @ ", url),
+                        None => String::new(),
+                    },
```

Comment:
If I'm being `option_if_let_else`:

```suggestion
                    info.get_repository_url().map_or_else(String::new, |url| format!("{} @ ", url))
```

`gir`/`gstreamer-rs` really need more strict lints turned on - if it wasn't so dreadful without a crate-shareable config file.

Reply:
Nice!